library/core/tests/time.rs

```diff
@@ -200,6 +200,7 @@ fn debug_formatting_extreme_values() {
         format!("{:?}", Duration::new(18_446_744_073_709_551_615, 123_456_789)),
         "18446744073709551615.123456789s"
     );
+    assert_eq!(format!("{:.0?}", Duration::MAX), "18446744073709551616s");
```

Comment:
Please add a few more boundary case tests here.  For example, something like
```rust
    assert_eq!(format!("{:.0?}", Duration::new(u64::MAX, 500_000_000)), "18446744073709551616s");
    assert_eq!(format!("{:.0?}", Duration::new(u64::MAX, 499_999_999)), "18446744073709551615s");
    assert_eq!(format!("{:.3?}", Duration::new(u64::MAX, 999_500_000)), "18446744073709551616.000s");
    assert_eq!(format!("{:.3?}", Duration::new(u64::MAX, 999_499_999)), "18446744073709551615.999s");
```
since it's not *just* `MAX` where this fails today.